rpmio/rpmkeyring.cc

```diff
@@ -35,6 +35,14 @@ struct rpmKeyring_s {
     std::shared_mutex mutex;
 };
 
+struct rpmKeyringIterator_s {
+    rpmKeyring keyring;
+    rdlock * keyringlock;
+    std::map<std::string,rpmPubkey>::const_iterator iterator;
+    rpmPubkey current;
+    std::shared_mutex mutex;
```

Comment:
The iterator itself is never shared, drop all this mutex stuff from here.

Reply:
You can have multiple threads with an iterator in each, but you can't share an iterator across multiple threads, that just doesn't make any sense.